crates/type-system/src/ontology/data_type/wasm.rs

```diff
@@ -0,0 +1,24 @@
+#![allow(
+    non_snake_case,
+    reason = "We want camelCase types in TS, and this module is WASM only"
+)]
+use wasm_bindgen::prelude::*;
+
+use crate::{ontology::data_type::error::MalformedDataTypeError, DataType};
+
+#[wasm_bindgen]
+extern "C" {
+    #[wasm_bindgen(typescript_type = "DataType")]
+    pub type IDataType;
+}
+
+/// Checks if a given {DataType} is valid
+///
+/// @throws {MalformedDataTypeError} if the data type is malformed
+#[wasm_bindgen]
+pub fn isValidDataType(dataTypeObj: &IDataType) -> Result<(), MalformedDataTypeError> {
```

Comment:
This pattern is probably worth explaining. 

- `Tsify` generates a TypeScript interface (see `derive` on `DataType`). 
- We then tell `wasm_bindgen` that there is an external type `IDataType` which it can find in the TypeScript bindings as `DataType`. 
- We can then in turn use that in the `wasm_bindgen` function here where `IDataType` becomes an alias for `JsValue`.
- This is important because it allows us to call `into_serde`, if we didn't rely on the binding and instead told it the real type, then it would do an unchecked_unwrap which will panic in WASM rather than returning serde errors.
- We need the `extern "C"` thing so that the wasm binding _for the functions_ have the correct interface on the function parameters